datafusion/physical-plan/src/aggregates/group_values/multi_group_by/bytes.rs

```diff
@@ -187,6 +194,13 @@ where
     {
         let value: &[u8] = array.value(row).as_ref();
         self.buffer.append_slice(value);
+
+        assert!(
```

Comment:
would it be better to return an error here to the user instead of a panic? I realize that would require some additional plumbing.

However, a panic is certainly better than SEGFAULT 👍

Reply:
Yeah I think in general it would be better. Panicking due to offset overflow is common in other places though so it's not that jarring :)

Reply:
https://github.com/apache/datafusion/issues/15969